### PR TITLE
LPS-201051 Research SPA style evaluation

### PR DESCRIPTION
**Pull request by chestofwonders**
Technical solution to https://liferay.atlassian.net/browse/LPS-201051
Confluence documentation: https://liferay.atlassian.net/wiki/x/JYCGm

### CODE REVIEW

**Review by markocikos — COMMENTED** [review]
@chestofwonders @dsanz I'm ok with merging this. It's not an ideal solution, but it looks to me like an improvement to what we currently have. I tested this a bit, navigating around DXP, and I didn't find any issues.

My only nit with this PR is that it is under the spike LPS. @chestofwonders you can create a technical subtask on the spike, send code under that LPS.

File: modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/app/App.js [hunk 1]
```diff
@@ -485,8 +485,8 @@ class App extends EventEmitter {
 					this.extractParams(route, path)
 				);
 			})
-			.then(() => nextScreen.evaluateStyles(this.surfaces))
 			.then(() => nextScreen.flip(this.surfaces))
+			.then(() => nextScreen.evaluateStyles(this.surfaces))
```
Comment (bryceosterhaus) [inline modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/app/App.js:489]: By flipping the screen before evaluating the styles, does that result in significant jank or layout shift when navigating between two pages that have a significant style difference?

I'm not entire sure how you would test this, but if you go from one page with say a white background to another page with a black background, will that delay in evaluating the styles cause a shift?

**Comment by chestofwonders** [comment]
PR opened under a technical subtask: https://github.com/liferay-frontend/liferay-portal/pull/3840

**Comment by markocikos** [comment]
Closing this, as it has a followup. @chestofwonders please see https://github.com/liferay-frontend/liferay-portal/pull/3834#discussion_r1398912628

File: modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/app/App.js (hunk 1 above)
Comment (chestofwonders) [reply modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/app/App.js:489]: As I tested it along the portal, it behaves better that before. Is there any special use case we should test too?

**Pull request closed without merging**